lib/logstash/outputs/syslog.rb

```diff
@@ -94,8 +94,12 @@ class LogStash::Outputs::Syslog < LogStash::Outputs::Base
   # syslog message format: you can choose between rfc3164 or rfc5424
   config :rfc, :validate => ["rfc3164", "rfc5424"], :default => "rfc3164"
 
+  # use codec to format message (instead of setting message)
+  config :usecodec, :validate => :boolean, :default => false
```

Comment:
I don't see how to implement this, because codec is never undefined. It allways defaults to plain because this is how it is set in https://github.com/elastic/logstash/blob/master/logstash-core/lib/logstash/outputs/base.rb#L20

I don't think, the following valid, but I didn't try it.

    default :codec, ""